### DISCUSSION TITLE
Problem Livewire hydrate

### DISCUSSION THREAD

**Discussion by EltlasTI** [livewire/livewire#8807] [7. Bugs (archived)]
### Livewire version

3.4

### Laravel version

10.48.10

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

Hi, in search or navigate record , show this message

Livewire encountered corrupt data when trying to hydrate a component. Ensure that the [name, id, data] of the Livewire component wasn't tampered with between requests.

this problem exists solution?

### Repository or Wirebox URL

https

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by PhiloNL** [comment]
Hi there,
Without any code, it's impossible to help. If you can replicate your issue using https://wirebox.app/ and share the URL that would be great.

**Reply by EltlasTI** [reply]
Hi, send the code my component and blade

//component

<?php
namespace App\Livewire\Legislacao;

use Livewire\Component;

use Livewire\WithPagination;
use Livewire\WithFileUploads;
use App\Models\vw_legislacao;
use App\Models\legislacao;
use App\Models\legislacao_origem;
use App\Models\legislacao_tipo;
use App\Models\legislacao_situacao;
use Carbon\Carbon;
use Carbon\Exceptions\InvalidFormatException;
use Illuminate\Support\Collection;

use Exception;
use Livewire\Exceptions\MethodNotFoundException;
use Psr\Container\NotFoundExceptionInterface;

class Legislacaos extends Component
{
   
    use WithPagination;
    use WithFileUploads;
    
    // estiliza a paginação para o formato bootstrap
    protected $paginationTheme = 'bootstrap';


     // variaveis da tabela
    public $registro, $data_criacao,$data_edicao,$usuario_criacao,$usuario_edicao,$ativo,$lixo,$legislacao_origem_reg,$doc_reg;
    public $legislacao_tipo_reg,$legislacao_situacao_reg,$data,$numero,$ementa,$integra,$observacao,$anexo,$arquivo,$visitas;
    
    // variavel para verificacao ordenação
    public $ord_aux=1;
    
    // variaveis de 
    public $filen,$filet;
    
    public $nometipo;

     // variaveis de pesquisa geral
     public $pesquisa;
     public $pesquisa1;

     protected $filtro;
     protected $filtro1;
 
     public $oppesquisa = 0;

 
    //  variaveis de ordenação    
    public $orderColumn = "ID";
    public $sortOrder = "asc";
    public $sortLink = '<i class="sorticon fa-solid fa-caret-up"></i>';

     // função de ordenação
    public function SortOrder($columnName="")
    {
         $caretOrder = "up";
         if($this->sortOrder == 'asc'){
              $this->sortOrder = 'desc';
              $caretOrder = "down";
         }else{
              $this->sortOrder = 'asc';
              $caretOrder = "up";
         } 
         $this->sortLink = '<i class="sorticon fa-solid fa-caret-'.$caretOrder.'"></i>';
 
         $this->orderColumn = $columnName;

         // seta para 2a. ordenação
         $this->ord_aux=2;
 
    }


    //function isValidDate($date, $format = 'Y-m-d') 
    function isValidDate($date, $format = 'd/m/Y') 
    {
        return Carbon::createFromFormat($format, $date, null, false) !== false;        
    }


    // atualiza opção de pesquisa selecionada
    public function updatingOppesquisa($value)
    {
        if ($value != $this->oppesquisa) 
        {
            $this->reset('pesquisa');
            $this->resetPage();
        } 
        elseif ($value == 0) 
        {
            $this->reset('pesquisa');
            $this->resetPage();
        } 
        else 
        {
            $this->resetPage();
        }
    }

    public function render()
    {

        try
        {
            // Seta ordenação geral por data e desc na 1a. vez                
            if ($this->ord_aux==1)
            {
                $this->sortOrder="desc";
                $this->orderColumn="DATA";
            }    
        
            // CARGA - FILTRO PAGINAÇÃO E ORDENAÇÃO POR TODOS OS CAMPOS | INICIO
            //$legislacaos = vw_legislacao::select('id','registro','data_criacao','doc_reg','numero','ementa','data','legislacao_tipo_reg','tipo')->orderBy($this->orderColumn,$this->sortOrder)->select('*');

            //$legislacaos = vw_legislacao::select('id','registro','data_criacao','doc_reg','numero','ementa','data','legislacao_tipo_reg','tipo')->orderBy('data','desc')->orderBy('numero','desc')->select('*');
			$legislacaos = vw_legislacao::select('id','registro','legislacao_origem_reg','data_criacao','doc_reg','numero','ementa','data','legislacao_tipo_reg','observacao')->orderBy('data','desc')->orderBy('numero','desc')->select('*');
            
            //$this->pesquisa= trim($this->pesquisa);
            //$this->pesquisa1= trim($this->pesquisa1);

            $this->filtro= trim($this->pesquisa);
            $this->filtro1= trim($this->pesquisa1);

            //* Verifica pesquisa inicio

            if ($this->pesquisa)
            {
            
                if ($this->oppesquisa == 1) 
                {
                
                    $this->filtro = intval($this->filtro);
                    if (is_int($this->filtro))
                    {
                        $legislacaos->where('legislacao_tipo_reg',$this->filtro);
                        //$this->reset('pesquisa');
                    }                           
                
                } else if ($this->oppesquisa == 2) 
            
                {
                    $this->filtro = intval($this->filtro);
                    if (is_int($this->filtro))
                    {
                        $legislacaos->where('numero',$this->filtro);
                        //$this->reset('pesquisa');
                    }
                } 
                else if ($this->oppesquisa == 3) 
                {
                    $legislacaos->where('EMENTA', 'like', '%'.$this->filtro.'%');
                    //$this->reset('pesquisa');
                } 


                else if ($this->oppesquisa == 4) 
                {
                    $legislacaos->where('INTEGRA', 'like', '%'.$this->filtro.'%');
                    //$this->reset('pesquisa');
                } 

                else if ($this->oppesquisa == 5) 
                {
                    $legislacaos->where('DATA', $this->filtro);
                    //$this->reset('pesquisa');
                } 


                else if ($this->oppesquisa == 6) 
                {
                    $legislacaos->where('legislacao_tipo_reg', $this->filtro)
                            ->where('numero', $this->filtro1);
                    //$this->reset('pesquisa');
                    //$this->reset('pesquisa1');
                } 


                else if ($this->oppesquisa == 7) 
                {
                    $legislacaos->where('legislacao_tipo_reg', $this->filtro)
                            ->where('data', $this->filtro1);
                    //$this->reset('pesquisa');
                    //$this->reset('pesquisa1');
                } 

                else if ($this->oppesquisa == 8) 
                {
                    $legislacaos->where('legislacao_tipo_reg', $this->filtro)
                                ->where('integra', 'like', '%'.$this->filtro1.'%');
                    //$this->reset('pesquisa');
                    //$this->reset('pesquisa1');
                }                             } 
                else 
                {
                    $this->reset('pesquisa');
                    $this->reset('pesquisa1');
                    
                } 

                //* verifica pesquisa fim
        
                $legislacaos = $legislacaos->paginate(5);

                if ($legislacaos->total()==0)
                {
                    session()->flash('messageerro', 'Nenhum registro encontrado.');

                }
        
                // seta para primeira página e retorna dados para view
        }
        catch (Exception $e)
        {
            session()->flash('messageerro', 'Ocorreu uma instabilidade/erro recarrague a página no seu navegador (F5)'.' | '.'Caso a instabilidade/erro persista envie um e-mail para sistemas@contagem.mg.gov.br ');
        }
                            
        // retorna para pagina numero 1
        $this->resetPage();
                         
        return view('livewire.legislacao.legislacaos',['legislacaos' => $legislacaos,]);

    }

    // Trata erro gerais
    public function exception($e, $stopPropagation) {
        if($e instanceof NotFoundExceptionInterface)
        {
            $this->notify('Erro');
            
        }
    }


    // função exclusão de registro
    public function delete($id)
    {
           legislacao::find($id)->delete();
           session()->flash('message', 'Legislação excluida com sucesso.');
    }

    // download de arquivo 
    //public function download($id)
    public function download($id,$file,$tp)
    { 

        try
        {
            // localiza id - arquivo        
            $legislacaos = vw_legislacao::findOrFail($id);

            if ($tp=='an')        
            {   
                $this->filet = $legislacaos->anexo; 

            }    
            else
            {
                $this->filet = $legislacaos->arquivo;
            }

            //$this->filen = public_path('storage/app/public/arquivos/legislacao/'.$this->arquivo);

            // pega nome completo do arquivo
            $this->filen = public_path('storage/arquivos/legislacao/'.$this->filet);

            // verifica existência do arquivo

            if(file_exists($this->filen)) 
            {
                return response()->download($this->filen);
            } 
            else 
            {
                session()->flash('messageerro', 'O arquvivo '.$this->filen.' não encontrado.');              
            }

        }    

          catch (Exception $e)
        {
            session()->flash('messageerro', 'Erro:  '.$e);
        }
        
    }

    // função lista tipos

    public function buscatipo($tipo)
    {                
        
        $tipo='092900';

        $legislacao_tipo = $tipo;

        $legislacoes = vw_Legislacao::with('tipo')->whereHas('tipo', function ($q) use ($legislacao_tipo) 
        {
            $q->where('tipo', $legislacao_tipo);

        })->get();

        

    }

}

// blade view

<!doctype html>
<div class="container">

        {{-- Mensagem de sucesso --}}

        @if (session()->has('message'))
            <div class="alert alert-success">
                {{ session('message') }}
            </div>
        @endif

        {{-- Mensagem de erro --}}
        
        @if (session()->has('messageerro'))
            <div class="alert alert-danger">
                {{ session('messageerro') }}
            </div>
        @endif


         

        
        <br>
        <br>
        <div class='row'>
            <div class='col'>   
                <button wire:navigate href="{{ route('legislacao.inclusao') }}" class="btn btn-dark">Incluir</button>                 
                <button wire:navigate href="{{ route('legislacao') }}" class="btn btn-secondary">Voltar</button>                 
            </div>    
        </div>
        <hr>
    
        {{-- opções de pesquisa inicio --}}

        <div class="row">
            <div class="container mb-3  text-center">
                <div class="col"> 
                    <div class="btn-group" role="group" aria-label="Basic radio toggle button group">
                                 
                        <input type="radio" class="btn-check" name="btnradio" id="btnradio0" value=0 autocomplete="off" wire:model.live='oppesquisa'>
                        <label class="btn btn-outline-success" for="btnradio0">Sem Filtro</label>
                    
                        <input type="radio" class="btn-check" name="btnradio" id="btnradio1" value=1 autocomplete="off" wire:model.live='oppesquisa'>
                        <label class="btn btn-outline-success" for="btnradio1">Tipo</label>

                        <input type="radio" class="btn-check" name="btnradio" id="btnradio2" value=2 autocomplete="off" wire:model.live='oppesquisa'>
                        <label class="btn btn-outline-success" for="btnradio2">Número</label>

                        <input type="radio" class="btn-check" name="btnradio" id="btnradio3" value=3 autocomplete="off" wire:model.live='oppesquisa'>
                        <label class="btn btn-outline-success" for="btnradio3">Ementa</label>

                        <input type="radio" class="btn-check" name="btnradio" id="btnradio4" value=4 autocomplete="off" wire:model.live='oppesquisa'>
                        <label class="btn btn-outline-success" for="btnradio4">Integra</label>

                        <input type="radio" class="btn-check" name="btnradio" id="btnradio5" value=5 autocomplete="off" wire:model.live='oppesquisa'>
                        <label class="btn btn-outline-success" for="btnradio5">Data</label>  

                        <input type="radio" class="btn-check" name="btnradio" id="btnradio6" value=6 autocomplete="off" wire:model.live='oppesquisa'>
                        <label class="btn btn-outline-success" for="btnradio6">Tipo e número</label>  

                        <input type="radio" class="btn-check" name="btnradio" id="btnradio7" value=7 autocomplete="off" wire:model.live='oppesquisa'>
                        <label class="btn btn-outline-success" for="btnradio7">Tipo e Data</label>  

                        <input type="radio" class="btn-check" name="btnradio" id="btnradio8" value=8 autocomplete="off" wire:model.live='oppesquisa'>
                        <label class="btn btn-outline-success" for="btnradio8">Tipo e Integra</label>  

                    </div>                        
                </div>
            </div>
        </div>    


            {{-- Adiciona uma linha de espaço entre elemento e outro --}}
            <div style="height:10px;"></div>

        <div class="row">
            <div class="container mb-3  text-center">        
                <div class="col">

                    @if ($oppesquisa)
                        <form wire:submit.prevent="pesquisa">
                
                            @if ($oppesquisa == 1)
                
                                <div class="text-center mb-3">                      
                                    <select name="SearchTipos" class="form-control-lg" wire:model="pesquisa" autocomplete="off">
                                        <option value="" > </option>
                                        <option value="092900" >Decreto</option>
                                        <option value="168789" >Decreto Legislativo</option>
                                        <option value="193640" >Emenda à Lei Orgânica</option>
                                        <option value="854013" >Lei</option>
                                        <option value="482620" >Lei Complementar</option>
                                        <option value="878312" >Lei Orgânica</option>
                                        <option value="816485" >Portaria</option>
                                        <option value="356708" >Razões de Veto</option>
                                        <option value="299856" >Resolução</option>                                            
                                    </select>            
                    
                                    <button class="btn btn-light" wire:click='$refresh'>Buscar</button>

                                    <div wire:loading wire:target="$refresh">  
                                        Removing post...
                                    </div>
                                </div>                     
                            @endif

                            @if ($oppesquisa == 2)
                                <div class="text-center mb-3">
                                    <input type="text" class="form-control-lg" placeholder="Número" step="1" min="1" max="99999999" wire:model="pesquisa" x-mask="999999999" autocomplete="off">
                                    <button class="btn btn-light" wire:click='$refresh'>Buscar</button>
                                </div>
                            @endif
                
                            @if ($oppesquisa == 3)
                                <div class="text-center mb-3">
                                    <input type="text" class="form-control-lg" placeholder="Digite seu texto" wire:model="pesquisa" autocomplete="off">
                                    <button class="btn btn-light" wire:click='$refresh' >Buscar</button>
                                </div>
                            @endif
                
                            @if ($oppesquisa == 4)
                                <div class="text-center mb-3">
                                    <input type="text" class="form-control-lg" placeholder="Digite seu texto" wire:model="pesquisa" autocomplete="off">
                                    <button class="btn btn-light" wire:click='$refresh'>Buscar</button>
                                </div>
                            @endif

                            @if ($oppesquisa == 5)
                                <div class="text-center mb-3">
                                    <input type="date" class="form-control-lg" placeholder="Data" wire:model="pesquisa" autocomplete="off">
                                    <button class="btn btn-light" wire:click='$refresh'>Buscar</button>
                                </div>
                            @endif


                            @if ($oppesquisa == 6)
                
                                <div class="text-center mb-3">                      
                                    <select name="SearchTipos" class="form-control-lg" wire:model="pesquisa" autocomplete="off">
                                        <option value="" > </option>
                                        <option value="092900" >Decreto</option>
                                        <option value="168789" >Decreto Legislativo</option>
                                        <option value="193640" >Emenda à Lei Orgânica</option>
                                        <option value="854013" >Lei</option>
                                        <option value="482620" >Lei Complementar</option>
                                        <option value="878312" >Lei Orgânica</option>
                                        <option value="816485" >Portaria</option>
                                        <option value="356708" >Razões de Veto</option>
                                        <option value="299856" >Resolução</option>                                            
                                    </select>        
                                    <label><b> &nbsp; E &nbsp; </b> </label>
                                    <input type="text" class="form-control-lg" placeholder="Digite o número" wire:model="pesquisa1" autocomplete="off">
                    
                                    <button class="btn btn-light" wire:click='$refresh'>Buscar</button>
                                </div>                     
                            @endif


                            @if ($oppesquisa == 7)
                
                                <div class="text-center mb-3">                      
                                    <select name="SearchTipos" class="form-control-lg" wire:model="pesquisa" autocomplete="off">
                                        <option value="" > </option>
                                        <option value="092900" >Decreto</option>
                                        <option value="168789" >Decreto Legislativo</option>
                                        <option value="193640" >Emenda à Lei Orgânica</option>
                                        <option value="854013" >Lei</option>
                                        <option value="482620" >Lei Complementar</option>
                                        <option value="878312" >Lei Orgânica</option>
                                        <option value="816485" >Portaria</option>
                                        <option value="356708" >Razões de Veto</option>
                                        <option value="299856" >Resolução</option>                                            
                                    </select>        
                                    <label><b> &nbsp; E &nbsp; </b> </label>
                                    <input type="date" class="form-control-lg" wire:model="pesquisa1" autocomplete="off">
                    
                                    <button class="btn btn-light" wire:click='$refresh'>Buscar</button>
                                </div>                     
                            @endif

                            @if ($oppesquisa == 8)
                
                                <div class="text-center mb-3">                      
                                    <select name="SearchTipos" class="form-control-lg" wire:model="pesquisa" autocomplete="off">
                                        <option value="" > </option>
                                        <option value="092900" >Decreto</option>
                                        <option value="168789" >Decreto Legislativo</option>
                                        <option value="193640" >Emenda à Lei Orgânica</option>
                                        <option value="854013" >Lei</option>
                                        <option value="482620" >Lei Complementar</option>
                                        <option value="878312" >Lei Orgânica</option>
                                        <option value="816485" >Portaria</option>
                                        <option value="356708" >Razões de Veto</option>
                                        <option value="299856" >Resolução</option>                                            
                                    </select>        
                                    <label><b> &nbsp; E &nbsp; </b> </label>
                                    <input type="text" class="form-control-lg" placeholder="Digite seu texto" wire:model="pesquisa1" autocomplete="off">
                    
                                    <button class="btn btn-light" wire:click='$refresh'>Buscar</button>
                                </div>                     
                            @endif
                        </form>
                    @endif
                </div>
            </div>
        </div>        

        {{-- opções de pesquisa fim --}}


               
                <p> <b> Para ordenação, clique no nome da coluna </b></p>

                <div class="row">
                    <div class="table-responsive-sm">
                        <table class="table table-responsive-sm table-bordered mt-5 table-Tertiary table-striped">
                            <thead>
                                <tr>                                  
                                    <th width="10px" class="sort" wire:click="SortOrder('id')">Tipo {!! $sortLink !!}</th>
                                    
                                    <th width="50px" class="sort" wire:click="SortOrder('numero')">Número {!! $sortLink !!}</th>
                                    <th width="50px" class="sort" wire:click="SortOrder('data')">Data {!! $sortLink !!}</th>                    
                                    <th width="550px" class="sort" wire:click="SortOrder('ementa')">Ementa {!! $sortLink !!}</th>
                                    <th width="70px" class="sort" wire:click="SortOrder('anexo')">Anexo {!! $sortLink !!}</th>
                                    <th width="70px" class="sort" wire:click="SortOrder('arquivo')">Arquivo {!! $sortLink !!}</th>
                                    <th width="210px" align="center">Ações</th>
                                </tr>
                            </thead>
                            <tbody>
                                @foreach($legislacaos as $legislacao)
                                    <tr>
                                        <td>{{ $legislacao->tipo }}</td>
                                        
                                        <td>{{ $legislacao->numero }}</td>
                                        
                                        
                                        <td>{{ date( 'd/m/Y' , strtotime($legislacao->data))}}</td>

                                        <td align="justify" style="height:100px;"> {!!html_entity_decode($legislacao->ementa)!!}  </td>
                            
                                        <td align="center">                
                                            
                                            
                                            @if ($legislacao->anexo)                                    
                                                <button type="button" class="btn btn-success btn-sm" wire:click="download('{{ $legislacao->id }}','{{$legislacao->anexo}}','an')" style="width:80px;">Download </button>
                                            @endif                                
                                        </td>    
                                        <td align="center">
                                                                                                                                          
                                            
                                            @if ($legislacao->arquivo)
                                                <button type="button" class="btn btn-success btn-sm" wire:click="download('{{ $legislacao->id }}','{{$legislacao->arquivo}}','ar')" style="width:80px;">Download </button>
                                            @endif                                
                                        </td>    
                                      
                                        <td>  
                                        
                                                                                    
                                            <a href="{{ route('legislacao.edicao', ['id' => $legislacao->id]) }}" class="btn btn-primary btn-sm" style="width:55px;">Editar</a>
                                            <button wire:click="delete({{ $legislacao->id }})" class="btn btn-danger btn-sm" style="width:55px;" style="width:55px;">Excluir</button>                                                                
                                        </td>
                                    </tr>
                                @endforeach           
                            </tbody>
        
                        </table>
                    </div>
                </div>    

                <div class="row">
                    <div class="col">    
                        
                        <div>
                            {{$legislacaos->links()}} 
                            <hr>
                            <label for="TotalREG"> Total de registros : {{$legislacaos->total()}} </label>       
                        </div>
                    </div>    
                </div>    
        {{-- Adiciona espaço para a barra --}}
        <div style="height:100px;"></div>
</div>


The error show after usualy option filter

**Comment by EltlasTI** [comment]
HI, send  too link code

https://wirebox.app/b/4nper
